examples/runc-hook/main.go

```diff
@@ -0,0 +1,222 @@
+		ID:      notif.ContainerID,
+		Pid:     int(notif.ContainerPID),
+		// TODO: Make runcfanotify return the bundle dir path too.
+		Bundle: "",
+	}
+	if notif.ContainerConfig != nil && notif.ContainerConfig.Annotations != nil {
+		ociState.Annotations = notif.ContainerConfig.Annotations
+	} else {
+		ociState.Annotations = make(map[string]string)
+	}
+
+	var cmd string
+	switch notif.Type {
+	case runcfanotify.EventTypeAddContainer:
+		ociState.Status = ocispec.StateCreated
+		var config string
+		if notif.ContainerConfig != nil {
+			b, err := json.MarshalIndent(notif.ContainerConfig, "", "  ")
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Failed to marshall ContainerConfig: %s\n", err)
+			} else {
+				config = string(b)
+			}
+		}
+		if outputAdd {
+			fmt.Printf("Container added: %v pid %d\n", notif.ContainerID, notif.ContainerPID)
+		}
+		if outputConfig {
+			if config != "" {
+				fmt.Printf("%s\n", config)
+			} else {
+				fmt.Fprintf(os.Stderr, "Error: container config not found for container %s\n", notif.ContainerID)
+			}
+		}
+		if *publishKubernetesEvent {
+			publishEvent("NewContainerConfig", config)
+		}
+
+		if *hookPreStart != "" {
+			cmd = *hookPreStart
+		}
+	case runcfanotify.EventTypeRemoveContainer:
+		ociState.Status = ocispec.StateStopped
+		if outputRemove {
+			fmt.Printf("Container removed: %v pid %d\n", notif.ContainerID, notif.ContainerPID)
+		}
+		if *hookPostStop != "" {
+			cmd = *hookPostStop
+		}
+	default:
+		return
+	}
+
+	if cmd != "" {
+		t, _ := time.ParseDuration(*timeout)
+		command := &configs.Command{
+			Path:    "/bin/sh",
+			Args:    []string{"/bin/sh", "-c", cmd},
+			Env:     strings.Split(*env, " "),
+			Dir:     *dir,
+			Timeout: &t,
+		}
+
+		// Run the command with a synthetised OCI state over stdin
+		err := command.Run(ociState)
+		if err != nil {
+			fmt.Printf("Error: %s\n", err)
+		}
+	}
+}
+
+func main() {
+	flag.Parse()
+
+	for _, o := range strings.Split(*outputList, ",") {
+		switch o {
+		case "add":
+			outputAdd = true
+		case "remove":
+			outputRemove = true
+		case "config":
+			outputConfig = true
+		case "":
+			// strings.Split() can generate empty strings
+		default:
+			fmt.Printf("invalid option: %q\n", o)
+			os.Exit(1)
+		}
+	}
+
+	if *publishKubernetesEvent {
+		var err error
+		if *kubeconfig == "" && os.Getenv("KUBECONFIG") != "" {
+			*kubeconfig = os.Getenv("KUBECONFIG")
+		}
+		client, err = k8sutil.NewClientset(*kubeconfig)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to get Kubernetes client set: %s\n", err)
+			os.Exit(1)
+		}
+	}
+
+	supported := runcfanotify.Supported()
+	if !supported {
+		fmt.Printf("runcfanotify not supported\n")
+		os.Exit(1)
+	}
+
+	var err error
+	notifier, err = runcfanotify.NewRuncNotifier(callback)
+	if err != nil {
+		fmt.Printf("runcfanotify failed: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Graceful shutdown
+	exit := make(chan os.Signal, 1)
+	signal.Notify(exit, os.Interrupt, syscall.SIGTERM)
+	<-exit
```

Comment:
This also highlights the lack of a "stop" method for the runcfanotify API

We are "lucky" in our use case here, as we are just killing the process here and never ever need to stop it before, but this should be improved for a generic library IMHO.

Again, out of scope for this PR. Just something probably to improve, as feedback from users of that lib :)